packages/server/src/routes/provider/alert/create.svelte

```diff
@@ -546,36 +587,105 @@
         <button
           class="delete"
           aria-label="close"
-          on:click={() => (isIconSelectorOpen = false)} />
+          on:click={() => {
+            isIconSelectorOpen = false;
+            isCustomIconSelectorOpen = false;
+          }} />
       </header>
-      <section class="modal-card-body">
-        <div class="icon-choiced-container">
-          <div class="icon-choices">
-            {#each Object.values(icons) as icon}
-              <div class="icon-container">
-                <button
-                  class="button is-large"
-                  class:has-background-info={tempIcon === icon}
-                  on:click={() => (tempIcon = icon)}>
-                  <span class="icon is-large">
-                    <i class={`${icon} fa-2x`} />
-                  </span>
-                </button>
+      {#if isCustomIconSelectorOpen}
+        <section class="modal-card-body">
+          <div class="icon-choiced-container">
+            {#if iconSearchResults}
```

Comment:
```suggestion
            {#if iconSearchTerm && iconSearchResults.length}
```
So if we keep `iconSearchResults` as an array, we can check both the term and the array's length to determine if we're supposed to have results. There may be a little extra work to do to handle an initial load where the user has not entered a search term nor do we have search results either.

Reply:
This ended up being a little silly? I'll see what you think haha

Reply:
The thing that makes it weird is not showing the error message until the user has actually hit enter for the first time

Reply:
Yep I was thinking you'd have to catch whether search was submitted but this looks good!